src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -1315,31 +1315,36 @@ private void ParseExtensibleDirective(in SyntaxListBuilder<RazorSyntaxNode> buil
 
                         var tokenDescriptor = descriptor.Tokens[i];
                         AcceptWhile(IsSpacingToken(includeNewLines: false, includeComments: true));
+                        var seenWhitespace = TokenBuilder.Count > 0;
```

Comment:
This seems like a rough way to determine if we saw whitespace. Why not just eat the whitespace above the token iterations? Aka when this `for (var i = 0; i < descriptor.Tokens.Count; i++)` loop starts it'd be at: `@inject |DateTime Foo`

Reply:
We do already eat the whitespace before entering the loop. We don't build it in to a node yet because we output that whitespace as CSharp or Markup based on what type of token comes next. Plus we have other places in this file where we use `TokenBuilder.Count > 0` to determine things. It doesn't seem unreliable to me. I'll add a `Debug.Assert(TokenBuilder.Count == 0)` before we eat the whitespace to provide some extra guarantee.